python/cuml/metrics/regression.pyx

```diff
@@ -140,47 +193,106 @@ def mean_squared_error(y_true, y_pred,
         A non-negative floating point value (the best value is 0.0), or an
         array of floating point values, one for each individual target.
     """
-    y_true, n_rows, n_cols, ytype = \
-        input_to_cuml_array(y_true, check_dtype=[np.float32, np.float64,
-                                                 np.int32, np.int64])
+    y_true, y_pred, sample_weight, multioutput, raw_multioutput = \
+        _prepare_input_reg(y_true, y_pred, sample_weight, multioutput)
 
-    y_pred, _, _, _ = \
-        input_to_cuml_array(y_pred, check_dtype=ytype,
-                            check_rows=n_rows, check_cols=n_cols)
+    return _mse(y_true, y_pred, sample_weight, multioutput, squared,
+                raw_multioutput)
 
-    if sample_weight is not None:
-        sample_weight, _, _, _ = \
-            input_to_cuml_array(sample_weight, check_dtype=ytype,
-                                check_rows=n_rows, check_cols=n_cols)
 
-    raw_multioutput = False
-    allowed_multioutput_str = ('raw_values', 'uniform_average',
-                               'variance_weighted')
-    if isinstance(multioutput, str):
-        if multioutput not in allowed_multioutput_str:
-            raise ValueError("Allowed 'multioutput' string values are {}. "
-                             "You provided multioutput={!r}"
-                             .format(allowed_multioutput_str, multioutput))
-        elif multioutput == 'raw_values':
-            raw_multioutput = True
-        elif multioutput == 'uniform_average':
-            # pass None as weights to np.average: uniform mean
-            multioutput = None
-    elif multioutput is not None:
-        multioutput, _, _, _ = \
-            input_to_cuml_array(multioutput, check_dtype=ytype)
-        if n_cols == 1:
-            raise ValueError("Custom weights are useful only in "
-                             "multi-output cases.")
+@with_cupy_rmm
+def mean_absolute_error(y_true, y_pred,
+                        sample_weight=None,
+                        multioutput='uniform_average'):
+    """Mean absolute error regression loss
 
-    # (y_true - y_pred) ** 2
-    output_errors = cp.subtract(y_true, y_pred)
-    output_errors = cp.multiply(output_errors, output_errors)
+    Be careful when using this metric with float32 inputs as the result can be
```

Comment:
maybe rephrase last sentence to: "float64 will have lower numerical error" rather than "we recommend to use float64" since float64 will be super slow on some cards

Reply:
Good point.